integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksTest.java

```diff
@@ -0,0 +1,204 @@
+
+import com.sequenceiq.cloudbreak.api.model.CredentialResponse;
+import com.sequenceiq.cloudbreak.api.model.PlatformResourceRequestJson;
+import com.sequenceiq.it.cloudbreak.newway.CloudbreakClient;
+import com.sequenceiq.it.cloudbreak.newway.CloudbreakTest;
+import com.sequenceiq.it.cloudbreak.newway.Credential;
+import com.sequenceiq.it.cloudbreak.newway.Networks;
+import com.sequenceiq.it.cloudbreak.newway.TestParameter;
+import com.sequenceiq.it.cloudbreak.newway.cloud.CloudProvider;
+import com.sequenceiq.it.cloudbreak.newway.cloud.CloudProviderHelper;
+import com.sequenceiq.it.cloudbreak.newway.cloud.OpenstackCloudProvider;
+import com.sequenceiq.it.util.LongStringGeneratorUtil;
+
+public class NetworksTest extends CloudbreakTest {
+
+    private static final String VALID_CRED_NAME = "valid-cred-";
+
+    private static final String INVALID_CRED_NAME = "invalid-cred";
+
+    private static final String INVALID_AV_ZONE = "invalid-avzone";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworksTest.class);
+
+    private String credentialName = "";
+
+    private Long credentialId = 0L;
+
+    private CloudProvider cloudProvider;
+
+    @Inject
+    private LongStringGeneratorUtil longStringGeneratorUtil;
+
+    public NetworksTest() {
+    }
+
+    public NetworksTest(CloudProvider cp, TestParameter tp) {
+        cloudProvider = cp;
+        setTestParameter(tp);
+    }
+
+    @BeforeTest ()
+    @Parameters("provider")
+    public void beforeTest(@Optional(OpenstackCloudProvider.OPENSTACK) String provider) {
+        LOGGER.info("before cluster test set provider: " + provider);
+        if (cloudProvider != null) {
+            LOGGER.info("cloud provider already set - running from factory test");
+            return;
+        }
+        cloudProvider = CloudProviderHelper.providerFactory(provider, getTestParameter())[0];
+    }
+
+    @BeforeTest()
+    public void createCredentialsAllProviders() throws  Exception {
+        credentialName = VALID_CRED_NAME + cloudProvider.getPlatform().toLowerCase() + "-nets";
+
+        given(CloudbreakClient.isCreated());
+        given(cloudProvider.aValidCredential()
+                .withName(credentialName)
+                .withDescription("")
+                .withCloudPlatform(cloudProvider.getPlatform()), credentialName + " credential is created.");
+        when(Credential.get(), credentialName + " credential has been got.");
+        then(Credential.assertThis(
+                (credential, t) -> Assert.assertEquals(credential.getResponse().getName(), credentialName)),
+                credentialName + " should be the name of the new credential."
+        );
+        getCredentialId();
+    }
+
+    @Test()
+    @Parameters({ "region" })
+    public void testGetNetworksWithCredName(@Optional("") String region) throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(Networks.request(createPlatformResourceRequest(null, credentialName, "", region)),
```

Comment:
For four arguments a builder like .withXxx(x) method chaining might be more readable